fairlearn/reductions/_moments/utility_parity.py

```diff
@@ -285,9 +312,13 @@ class TruePositiveRateParity(UtilityParity):
     def load_data(self, X, y, **kwargs):
         """Load the specified data into the object."""
         # The `where` clause is used to put `pd.nan` on all values where `Y!=1`.
-        super().load_data(X, pd.Series(y),
-                          event=pd.Series(y).apply(lambda y: _LABEL + "=" + str(y)).where(y == 1),
-                          **kwargs)
+        base_event = pd.Series(y).apply(lambda y: _LABEL + "=" + str(y)).where(y == 1)
+        control_features = kwargs.get(_KW_CONTROL_FEATURES)
+        if control_features is not None:
+            event = base_event.combine(pd.Series(control_features), _combine_event_and_control)
+        else:
+            event = base_event
+        super().load_data(X, pd.Series(y), event=event, **kwargs)
```

Comment:
There's too much copy-pasting happening here. Also `**kwargs` doesn't help. We should have something like
```python
    def load_data(self, X, y, *, sensitive_features, control_features=None):
```

Reply:
There has been some heavy refactoring of this code. Let me know what you think.